applications/GeoMechanicsApplication/custom_elements/compressibility_calculator.cpp

```diff
@@ -0,0 +1,87 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#include "compressibility_calculator.h"
+#include "custom_utilities/transport_equation_utilities.hpp"
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+CompressibilityCalculator::CompressibilityCalculator(InputProvider rInputProvider)
+    : mInputProvider(std::move(rInputProvider))
+{
+}
+
+Matrix CompressibilityCalculator::LHSContribution()
+{
+    return LHSContribution(CalculateCompressibilityMatrix());
+}
+
+Vector CompressibilityCalculator::RHSContribution()
+{
+    return RHSContribution(CalculateCompressibilityMatrix());
+}
+
+Vector CompressibilityCalculator::RHSContribution(const Matrix& rCompressibilityMatrix) const
+{
+    return -prod(rCompressibilityMatrix, mInputProvider.GetNodalValues(DT_WATER_PRESSURE));
+}
+
+Matrix CompressibilityCalculator::LHSContribution(const Matrix& rCompressibilityMatrix) const
+{
+    return rCompressibilityMatrix * mInputProvider.GetDtPressureCoefficient();
```

Comment:
The DtPressureCoefficient is determined by the time integration scheme. I wonder if we should do this multipication inside the calculator of leave this outside to avoid the mingling of functionality in the calculator.